cloud/openstack/deployer.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+type DeploymentClient struct{}
+
+func NewDeploymentClient() *DeploymentClient {
+	return &DeploymentClient{}
+}
+
+func (*DeploymentClient) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (string, error) {
+	if machine.ObjectMeta.Annotations != nil {
+		if ip, ok := machine.ObjectMeta.Annotations[OpenstackIPAnnotationKey]; ok {
+			glog.Infof("Returning IP from machine annotation %s", ip)
+			return ip, nil
+		}
+	}
+
+	return "", errors.New("could not get IP")
+}
+
+func (d *DeploymentClient) GetKubeConfig(cluster *clusterv1.Cluster, master *clusterv1.Machine) (string, error) {
+	ip, err := d.GetIP(cluster, master)
+	if err != nil {
+		return "", err
+	}
+
+	result := strings.TrimSpace(util.ExecCommand(
+		"ssh", "-i", "/root/.ssh/openstack_tmp",
```

Comment:
how does this file end up in the container where the machine-controller is running?

Reply:
This GetKubeConfig() is only for clusterctl bootstrap the cluster. But still, use this absolute path is not a good solution. 
I think this whole `DeploymentClient` can be removed after this issue finished. https://github.com/kubernetes-sigs/cluster-api/issues/157